lib/record_store/zone/config/ignore_pattern.rb

```diff
@@ -0,0 +1,36 @@
+module RecordStore
+  class Zone
+    class Config
+      class IgnorePattern
+        attr_accessor :orig_hash
+        alias_method :to_hash, :orig_hash
+
+        def initialize(orig_hash)
+          @orig_hash = orig_hash
+        end
+
+        def should_ignore?(record)
+          all_pairs.all?{|(key, value)| record.respond_to?(key) && value_matches?(value, record.send(key))}
+        end
+
+        def all_pairs
+          orig_hash.reject{|key, value| key === 'match'}
+        end
+
+        def value_matches?(ignore_pattern_value, record_value)
+          return record_value.match?(ignore_pattern_value) if is_regex?
+          return record_value === ignore_pattern_value if is_exact?
+          false
+        end
+
+        def is_regex?
+          orig_hash['match'] === 'regex'
```

Comment:
Any real special reason on using "===" rather than "==". Isn't that too defensive?

Reply:
Nitpick... we could use constants for the two strings `regex` and `exact`.